lib/fizzy/execute.cpp

```diff
@@ -187,42 +190,41 @@ std::tuple<bytes_ptr, size_t> allocate_memory(const std::vector<Memory>& module_
 
     if (module_memories.size() == 1)
     {
-        const size_t memory_min = module_memories[0].limits.min;
-        const size_t memory_max =
-            (module_memories[0].limits.max.has_value() ? *module_memories[0].limits.max :
-                                                         MemoryPagesLimit);
+        const auto memory_min = module_memories[0].limits.min;
+        const auto memory_max = module_memories[0].limits.max;
```

Comment:
The execution logic of `memory.grow` should stay the same, but for unlimited memory we don't save `MemoryPagesLimit` in `Instance` (we save only original module's or impored memory limits). For unlimited memory `MemoryPagesLimit` is enforced in `memory.grow` implemetation.